### ISSUE TITLE
mutually exclusive group w/ all parameters argparse.SUPPRESS'd causes an assert

### ISSUE THREAD

**Issue by kenyee** [python/cpython#98666]
# Bug report

```python
state_group = parser.add_mutually_exclusive_group()
state_group.add_argument('--param1', nargs='?', const='default', metavar='NAME',
                         help=SUPPRESS)
state_group.add_argument('--param2', nargs='?', const='default', metavar='NAME',
                         help=SUPPRESS)
```

will cause an assert in 
  `assert ' '.join(opt_parts) == opt_usage`
of argparse.py

because the empty group [] is converted into two spaces
e.g.
[group1] [] [group 2]
gets converted to this by the `opt_usage = format(optionals, groups)` code above the assert:
[group1]  [group 2]
and the two spaces don't match the single space of the assert

# Your environment

- CPython versions tested on: 3.10.4
- Operating system and architecture: OSX, arm64

**Comment by hpaulj** [comment]
The usage formatting is known for being brittle, especially when working with the extra notation of mutually_exclusive_groups.  This assert was intended to catch formatting errors that would cause problems if the usage has to be split onto several lines.  Without getting into the details, I'm not surprised that a rarely used feature like SUPPRESS causes problems.

**Comment by encukou** [comment]
I cannot reproduce with the above reproducer, which I expanded to:
```py
import argparse
from argparse import SUPPRESS

parser = argparse.ArgumentParser(prog='PROG')
state_group = parser.add_mutually_exclusive_group()
state_group.add_argument('--param1', nargs='?', const='default', metavar='NAME',
                         help=SUPPRESS)
state_group.add_argument('--param2', nargs='?', const='default', metavar='NAME',
                         help=SUPPRESS)
parser.parse_args(['--help'])
```
This is giving me the output:
```
usage: PROG [-h]

optional arguments:
  -h, --help  show this help message and exit
```
on Python versions 3.6 to 3.13.

If you have a complete example of failing behavior (even on an old version of Python), please share it here.


I think this problem was fixed in #105039; I'll close the issue.
